exchanges/kraken/kraken_websocket.go

```diff
@@ -286,303 +289,249 @@ func (k *Kraken) wsProcessSystemStatus(respRaw []byte) error {
-					Fee:                  val.Fee,
-					Date:                 val.OpenTime.Time(),
-					LastUpdated:          val.LastUpdated.Time(),
-				}
-
-				if val.Status != "" {
-					if s, err := order.StringToOrderStatus(val.Status); err != nil {
+			if val.Description.Pair != "" {
+				if strings.Contains(val.Description.Order, "sell") {
+					d.Side = order.Sell
+				} else {
+					if oSide, err := order.StringToOrderSide(val.Description.Type); err != nil {
 						k.Websocket.DataHandler <- order.ClassificationError{
 							Exchange: k.Name,
 							OrderID:  key,
 							Err:      err,
 						}
 					} else {
-						d.Status = s
+						d.Side = oSide
 					}
 				}
 
-				if val.Description.Pair != "" {
-					if strings.Contains(val.Description.Order, "sell") {
-						d.Side = order.Sell
-					} else {
-						if oSide, err := order.StringToOrderSide(val.Description.Type); err != nil {
-							k.Websocket.DataHandler <- order.ClassificationError{
-								Exchange: k.Name,
-								OrderID:  key,
-								Err:      err,
-							}
-						} else {
-							d.Side = oSide
-						}
+				if oType, err := order.StringToOrderType(val.Description.OrderType); err != nil {
+					k.Websocket.DataHandler <- order.ClassificationError{
+						Exchange: k.Name,
+						OrderID:  key,
+						Err:      err,
 					}
+				} else {
+					d.Type = oType
+				}
 
-					if oType, err := order.StringToOrderType(val.Description.OrderType); err != nil {
-						k.Websocket.DataHandler <- order.ClassificationError{
-							Exchange: k.Name,
-							OrderID:  key,
-							Err:      err,
-						}
-					} else {
-						d.Type = oType
+				if p, err := currency.NewPairFromString(val.Description.Pair); err != nil {
+					k.Websocket.DataHandler <- order.ClassificationError{
+						Exchange: k.Name,
+						OrderID:  key,
+						Err:      err,
 					}
-
-					if p, err := currency.NewPairFromString(val.Description.Pair); err != nil {
+				} else {
+					d.Pair = p
+					if d.AssetType, err = k.GetPairAssetType(p); err != nil {
 						k.Websocket.DataHandler <- order.ClassificationError{
 							Exchange: k.Name,
 							OrderID:  key,
 							Err:      err,
 						}
-					} else {
-						d.Pair = p
-						if d.AssetType, err = k.GetPairAssetType(p); err != nil {
-							k.Websocket.DataHandler <- order.ClassificationError{
-								Exchange: k.Name,
-								OrderID:  key,
-								Err:      err,
-							}
-						}
 					}
 				}
+			}
 
-				if val.Description.Price > 0 {
-					d.Leverage = val.Description.Leverage
-					d.Price = val.Description.Price
-				}
+			if val.Description.Price > 0 {
+				d.Leverage = val.Description.Leverage
+				d.Price = val.Description.Price
+			}
 
-				if val.Volume > 0 {
-					// Note: We don't seem to ever get both there values
-					d.RemainingAmount = val.Volume - val.ExecutedVolume
-				}
-				k.Websocket.DataHandler <- d
+			if val.Volume > 0 {
+				// Note: We don't seem to ever get both there values
```

Comment:
🚧 Needs rewording. I'm guessing at the meaning here:
```suggestion
				// Note: Seems we only ever receive RemainingAmount or Volume, never both
```

🤷 But I'm wondering why we're not checking for RemainingAmount